src/main/java/com/example/dbdemo/service/CourseServiceImpl.java

```diff
@@ -24,24 +27,35 @@ public Course getCourseById(Long id) {
     }
 
     @Override
-    public Course createCourse(Course course) {
+    public List<Course> getCoursesByIds(@NotNull List<Long> ids) {
+        if (ids.isEmpty())
+            return courseRepository.findAll();
+
+        return courseRepository.findAllById(ids);
+    }
+
+    @Override
+    public Course createCourse(@Valid Course course) {
         return courseRepository.save(course);
     }
 
     @Override
-    public Course updateCourse(Long id, Course course) {
+    public Course updateCourse(@NotNull @Min(1) Long id, @Valid Course course) {
```

Comment:
1. put `@valid` annotation in the controller (also check for the other API(s) as well)
2. No need to take  id explicitly as it is already there in the request payload